django/http/request.py

```diff
@@ -89,14 +90,47 @@ def headers(self):
 
     @cached_property
     def accepted_types(self):
-        """Return a list of MediaType instances."""
-        return parse_accept_header(self.headers.get("Accept", "*/*"))
+        """Return a list of MediaType instances, in order of preference."""
+        header_value = self.headers.get("Accept", "*/*")
+        return sorted(
+            (MediaType(token) for token in header_value.split(",") if token.strip()),
+            key=operator.attrgetter("quality", "specificity"),
+            reverse=True,
+        )
 
-    def accepts(self, media_type):
-        return any(
-            accepted_type.match(media_type) for accepted_type in self.accepted_types
+    def accepted_type(self, media_type):
+        return next(
+            (
+                accepted_type
+                for accepted_type in self.accepted_types
+                if accepted_type.match(media_type)
+            ),
+            None,
         )
 
+    def get_preferred_format(self, formats):
```

Comment:
Is this _the correct name_? I mean specifically `..._format` (method name on the `request`) and `formats` (argument) are somewhat ambiguous, and the other parts talk about _media types_ and _accepts_ (also in the case of the documentation the nomenclature is either mime type or content type) ...

I'm not making any judgement here, just pointing it out.

Reply:
This method returns a single _format_ given a set of _formats_. I agree though it'd be good to standardize. Perhaps "media type" is the correct name.